### ISSUE TITLE
bool_comparison causes type error 

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#9907]
### Summary

.

### Lint Name

bool_comparison

### Reproducer

I tried this code:

```rust
type Array = [(); ((1 < 2) == false) as usize];

fn main() {
    let _: Array = [];
}
```

I saw this happen:
`cargo clippy --fix -- -Aclippy::all -Wclippy::bool_comparison`
```
warning: failed to automatically apply fixes suggested by rustc to crate `clpy`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0308]: mismatched types
 --> src/main.rs:1:19
  |
1 | type Array = [(); 1 >= 2 as usize];
  |                   ^^^^^^^^^^^^^^^ expected `usize`, found `bool`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
Original diagnostics will follow.

warning: equality checks against false can be replaced by a negation
 --> src/main.rs:1:19
  |
1 | type Array = [(); ((1 < 2) == false) as usize];
  |                   ^^^^^^^^^^^^^^^^^^ help: try simplifying it as shown: `1 >= 2`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#bool_comparison
  = note: requested on the command line with `-W clippy::bool-comparison`

warning: `clpy` (bin "clpy") generated 1 warning (run `cargo fix --bin "clpy"` to apply 1 suggestion)
warning: `clpy` (bin "clpy" test) generated 1 warning (1 duplicate)
    Finished dev [unoptimized + debuginfo] target(s) in 0.88s
```

I expected to see this happen:

### Version

```text
rustc 1.67.0-nightly (c5d82ed7a 2022-11-19)
binary: rustc
commit-hash: c5d82ed7a4ad94a538bb87e5016e7d5ce0bd434b
commit-date: 2022-11-19
host: x86_64-unknown-linux-gnu
release: 1.67.0-nightly
LLVM version: 15.0.4
```
